zerver/views/user_groups.py

```diff
@@ -23,6 +23,10 @@ def add_user_group(request: HttpRequest, user_profile: UserProfile,
                    name: str=REQ(),
                    members: List[int]=REQ(validator=check_list(check_int), default=[]),
                    description: str=REQ()) -> HttpResponse:
+    realm = user_profile.realm
+    if realm.only_admins_create_user_groups and not user_profile.is_realm_admin:
+        raise JsonableError(_("Must be an organization administrator"))
```

Comment:
Rather than copying this code in 4 functions, I think we should do these checks inside a decorator at the top of this file replacing `@require_member_or_admin` that does this check as well.